lib/api/src/externals/function.rs

```diff
@@ -67,6 +67,49 @@ pub struct Function {
     pub(crate) exported: ExportFunction,
 }
 
+fn build_export_function_metadata<Env>(
+    env: Env,
+    import_init_function_ptr: for<'a> fn(
+        &'a mut Env,
+        &'a crate::Instance,
+    ) -> Result<(), crate::HostEnvInitError>,
+) -> (*mut std::ffi::c_void, ExportFunctionMetadata)
+where
+    Env: Clone + Sized + 'static + Send + Sync,
+{
+    let import_init_function_ptr = Some(unsafe {
+        std::mem::transmute::<fn(_, _) -> Result<(), _>, fn(_, _) -> Result<(), _>>(
+            import_init_function_ptr,
+        )
+    });
+    let host_env_clone_fn: fn(*mut std::ffi::c_void) -> *mut std::ffi::c_void = |ptr| {
+        let duped_env: Env = unsafe {
+            let ptr: *mut Env = ptr as _;
+            let item: &Env = &*ptr;
+            item.clone()
+        };
+        Box::into_raw(Box::new(duped_env)) as _
+    };
+    let host_env_drop_fn: fn(*mut std::ffi::c_void) = |ptr: *mut std::ffi::c_void| {
+        unsafe { Box::from_raw(ptr as *mut Env) };
+    };
```

Comment:
Just for the sake of consistency with my previous proposal, and for more simplicity:

```suggestion
    let host_env_drop_fn: fn(*mut std::ffi::c_void) = |ptr| {
        unsafe { Box::from_raw(ptr.cast::<Env>()) };
    };
```